src/exec/mod.rs

```diff
@@ -91,26 +121,64 @@ impl CommandTemplate {
             args.push(ArgumentTemplate::Tokens(vec![Token::Placeholder]));
         }
 
-        CommandTemplate { args }
+        CommandTemplate { args, mode }
+    }
+
+    fn tokens_number(&self) -> usize {
```

Comment:
Can we rename this to `number_of_tokens` or something similar? It sounds like it returns the number of a specific token.